### PR TITLE
[styled-system] Added missing grid functions, props & values

### PR DESCRIPTION
**Pull request by maoueh**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jxnblk/styled-system/blob/master/docs/table.md#grid-layout
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by sandersn** [comment]
@maxdeviant
@phobon
@zephraph
@damassi
@alloy
@maoueh

@typescript-bot failed to ping the owners here. Can one of you take a look?

File: types/styled-system/index.d.ts
Comment (phobon) [inline types/styled-system/index.d.ts]: Grid auto flow is actually a union type, so you could write it like:

```
export type GridAutoFlowValue =
    "row"
    | "column"
    | "dense"
    | "row dense"
    | "column dense"
    | "inherit"
    | "initial"
    | "unset";
```

File: types/styled-system/index.d.ts
Comment (phobon) [inline types/styled-system/index.d.ts]: With the change to GridAutoFlowValue above, this can stay as is - a string

File: types/styled-system/index.d.ts
Comment (phobon) [inline types/styled-system/index.d.ts]: This can be a ``` string | number``` right?

**Review by phobon — CHANGES_REQUESTED** [review]
Couple of minor changes there, but otherwise 👍

**Comment by zephraph** [comment]
I don't actually know why it's saying it's unassigned. The comments in the header seem to be correct. I cross referenced them with another project.

File: types/styled-system/index.d.ts
Comment (alloy) [inline types/styled-system/index.d.ts]: Are you aware that this means signing up as a maintainer and that you’ll get pinged on future PRs and tickets (once the bot starts working again, that is)?

File: types/styled-system/index.d.ts
Comment (alloy) [reply types/styled-system/index.d.ts]: @zephraph we should start using that CSS styles dependency that was mentioned on the first PR asap to set a precedent for people to follow.

**Review by just-be-dev — APPROVED** [review]

**Comment by alloy** [comment]
@maoueh FYI Waiting to hear re @phobon’s feedback before I’ll merge this.

**Comment by alloy** [comment]
@phobon @zephraph If the current state of the PR is at least a step up, then let’s just merge and improve the next time one of us uses these types?

**Comment by phobon** [comment]
Yeah, that sounds good to me

**Review by phobon — APPROVED** [review]
Approving these for now 👍

File: types/styled-system/index.d.ts
Comment (maoueh) [reply types/styled-system/index.d.ts]: Yep aware, but thanks for noticing it :)

**Pull request merged**
